testConfig.py

```diff
@@ -301,6 +325,16 @@ def parse(
         else:
             secondary_network_nad = v
 
+        v = vdict.pop("resource_name", None)
+        if v is None:
+            resource_name = get_resource_name()
```

Comment:
`get_resource_name()` has vast side effects, fetching the value from the current cluster.

While some attributes in `testConfig.py` are computed and don't directly come from the YAML (e.g. `ConfPlugin.plugin`), I think it's important that `TestConfig` primarily represents the content from the YAML file with only minor (stateless, side-effect-free) additions.

This means, make `resource_name: Optional[str]`, and leave it at None if it's unspecified in the YAML. Then, the caller who uses this field (`Task.get_template_args()`) has to resolve the default value, like

```python
    def get_template_args(self) -> dict[str, str | list[str]]:
        return {
            "name_space": self.get_namespace(),
            ...
            "secondary_network_nad": self.ts.connection.secondary_network_nad,
            "resource_name": self.ts.connection.resource_name or _fetch_default_resource_name(),
        }
```

---

Theoretically, we could also add a `ConfConnection.get_real_resource_name(self)->str:` method, that resolves the default value. That would be OK, since the data class still represents purely the YAML with an additional utility getter. But since there is only one caller (`Task.get_template_args()`), let `Task` handle this.

---

If you make `resource_name: Optional[str]`, then in `ConfConnection.serialize()`, maybe:
```diff
     def serialize(self) -> dict[str, Any]:
+        extra: dict[str, Any] = {}
+        common.dict_add_optional(extra, "resource_name", self.resource_name)
         return {
             **super().serialize(),
             "type": self.test_type.name,
             "instances": self.instances,
             "server": [s.serialize() for s in self.server],
             "client": [c.serialize() for c in self.client],
             "plugins": [p.serialize() for p in self.plugins],
             "secondary_network_nad": self.secondary_network_nad,
-            "resource_name": self.resource_name,
+            **extra,
         }
```

Reply:
Okay I can try to do it the first way! Thanks :)